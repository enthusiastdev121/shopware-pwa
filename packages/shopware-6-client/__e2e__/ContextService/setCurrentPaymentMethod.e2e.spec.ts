import {
  setCurrentPaymentMethod,
  update,
} from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - ContextService - setCurrentPaymentMethod", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should set 'Cash on Delivery' payment method", async () => {
    const result = await setCurrentPaymentMethod(
      "cad2356ee68c473ca7da517d0ad32de1"
    );
    result.contextToken = "mockedContextToken";
    expect(result).toMatchSnapshot();
  });

  it("should get error response when not existing payment method", async () => {
    try {
      await setCurrentPaymentMethod("qwe");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
