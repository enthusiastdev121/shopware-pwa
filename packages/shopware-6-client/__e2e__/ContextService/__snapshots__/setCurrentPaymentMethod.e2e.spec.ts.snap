// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - ContextService - setCurrentPaymentMethod should get error response when not existing payment method 1`] = `[Error: Request failed with status code 400]`;

exports[`shopware-6-client - E2E - ContextService - setCurrentPaymentMethod should set 'Cash on Delivery' payment method 1`] = `
Object {
  "contextToken": "mockedContextToken",
}
`;
